### PR TITLE
Update deprecated enums in rviz_common

### PR DESCRIPTION
**Pull request by brawner**
Several uses of enum values in rviz_common use long-deprecated values (back to 5.9). Compiling with 5.14.1 now produce deprecation warnings. These values have been mapped for awhile to the replacements I dropped in here, so this won't affect functionality.

`QPalette::Background` -> `QPalette::Window`
https://doc.qt.io/qt-5/qpalette.html#ColorRole-enum
`Qt::TextColorRole` -> `Qt::ForegroundRole`
https://doc.qt.io/qt-5/qt.html#ItemDataRole-enum

Addresses issue  #508

Build with warnings:
https://ci.ros2.org/job/ci_windows-container/123/warnings43Result/

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9393/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5072/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7687/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9311/)
* Windows-container [Build Status](http://ci.ros2.org/job/ci_windows-container/174/)

### CODE REVIEW

**Review by dirk-thomas — APPROVED** [review]

**Comment by brawner** [comment]
@dirk-thomas CI builds passed. Does this need a second review?

**Comment by dirk-thomas** [comment]
> Does this need a second review?

No, you can go ahead and merge.

**Comment by dirk-thomas** [comment]
@brawner I assume this can be merged? I will go ahead and do so.

**Pull request merged**
